src/components/ClusterOverview/Capacity/Capacity.js

```diff
@@ -15,56 +15,72 @@ const pfGetUtilizationDonutTooltipContents = d =>
-          const available = stats[key].data.total - stats[key].data.used;
-          const columns = [['Used', stats[key].data.used], ['Available', available]];
-          const sum = columns.reduce((acc, x) => acc + x[1], 0);
-          return (
-            <div key={key} className="kubevirt-capacity__item">
-              <h2 className="kubevirt-capacity__item-title">{stats[key].title}</h2>
-              <h6>
-                {`${Number.isInteger(available) ? available : available.toFixed(1)}${
-                  stats[key].unit
-                } available out of ${stats[key].data.total}${stats[key].unit}`}
-              </h6>
-              <div>
-                <DonutChart
-                  id={`donut-chart-${key}`}
-                  size={{ width: 150, height: 150 }}
-                  data={{
-                    unload: true,
-                    columns,
-                    order: null,
-                  }}
-                  unloadBeforeLoad
-                  tooltip={{ contents: pfGetUtilizationDonutTooltipContents }}
-                  title={{
-                    primary: `${Math.round((100 * columns[0][1]) / sum).toString()}%`,
-                    secondary: columns[0][0],
-                  }}
-                />
-              </div>
-            </div>
-          );
-        })}
+const CapacityItem = ({ idSuffix, title, used, total, unit }) => {
+  const available = total - used;
+  const columns = [['Used', used], ['Available', available]];
+
+  const sum = columns.reduce((acc, x) => acc + x[1], 0);
+  const percentageUsed = Math.round((100 * columns[0][1]) / sum);
+
+  const availableNice = Number.isInteger(available) ? available : available.toFixed(1);
+  const totalNice = Number.isInteger(total) ? total : total.toFixed(1);
+
+  return (
+    <div className="kubevirt-capacity__item">
+      <h2 className="kubevirt-capacity__item-title">{title}</h2>
+      <h6>{`${availableNice} ${unit} available out of ${totalNice} ${unit}`}</h6>
+      <div>
+        <DonutChart
+          id={`donut-chart-${idSuffix}`}
+          size={{ width: 150, height: 150 }}
+          data={{
+            unload: true,
+            columns,
+            order: null,
+          }}
+          unloadBeforeLoad
+          tooltip={{ contents: pfGetUtilizationDonutTooltipContents }}
+          title={{
+            primary: `${percentageUsed.toString()}%`,
+            secondary: columns[0][0],
+          }}
+        />
       </div>
-    );
-  }
-}
+    </div>
+  );
+};
+CapacityItem.propTypes = {
+  idSuffix: PropTypes.string.isRequired,
+  title: PropTypes.string.isRequired,
+  used: PropTypes.number.isRequired,
+  total: PropTypes.number.isRequired,
+  unit: PropTypes.string.isRequired,
+};
 
+const CapacityBody = ({ stats }) => (
+  <div className="kubevirt-capacity__items">
+    {Object.keys(stats)
+      .filter(key => stats[key].data && !Number.isNaN(stats[key].data.used) && !Number.isNaN(stats[key].data.total))
```

Comment:
So now you do not show CapacityItem if it does not have data yet. I would suggest to show title and under it show loading. IMO it will look better - user knows what will be shown right away..CPU/MEMORY/STORAGE/NETWORK and we will also prevent these items jumping around.

Reply:
done, for screenshots see https://github.com/kubevirt/web-ui/pull/235#issuecomment-475223232